apps/gateway/lib/gateway/controllers/curse_of_mirra/user_controller.ex

```diff
@@ -10,6 +10,18 @@ defmodule Gateway.Controllers.CurseOfMirra.UserController do
 
   action_fallback Gateway.Controllers.FallbackController
 
+  def show(conn, %{"id" => user_id}) do
+    game_id = Utils.get_game_id(:curse_of_mirra)
+
+    with {:get_user, {:ok, user}} <- {:get_user, Users.get_user(user_id)},
+         {:curse_user, ^game_id} <- {:curse_user, user.game_id} do
+      send_resp(conn, 200, Jason.encode!(user))
+    else
+      {:get_user, _} -> send_resp(conn, 404, "User not found")
+      {:curse_user, _} -> send_resp(conn, 400, "User from another game")
```

Comment:
Don't do this, use the `FallbackController` and handle things there

References:
- https://hexdocs.pm/elixir/code-anti-patterns.html#complex-else-clauses-in-with
- https://hexdocs.pm/phoenix/json_and_apis.html#action-fallback

Reply:
Done! thanks for the references